plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/TrinoNessieCatalog.java

```diff
@@ -0,0 +1,450 @@
+        this.nessieClient = requireNonNull(nessieClient, "nessieClient is null");
+    }
+
+    @Override
+    public List<String> listNamespaces(ConnectorSession session)
+    {
+        try {
+            return nessieClient.getNessieApi().getMultipleNamespaces()
+                    .refName(nessieClient.getRef().getName())
+                    .get().getNamespaces().stream()
+                    .map(Namespace::name)
+                    .collect(Collectors.toList());
+        }
+        catch (NessieReferenceNotFoundException e) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR,
+                    "Cannot list Namespaces: ref is no longer valid", e);
+        }
+    }
+
+    @Override
+    public void dropNamespace(ConnectorSession session, String namespace)
+    {
+        try {
+            nessieClient.getRef().checkMutable();
+            nessieClient.getNessieApi().deleteNamespace()
+                    .refName(nessieClient.getRef().getName())
+                    .namespace(namespace)
+                    .delete();
+            propertiesByNamespace.remove(namespace);
+            nessieClient.refreshReference();
+        }
+        catch (NessieNamespaceNotFoundException e) {
+            throw new SchemaNotFoundException(namespace);
+        }
+        catch (NessieReferenceNotFoundException e) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR,
+                    format("Cannot drop Namespace '%s': ref is no longer valid", namespace),
+                    e);
+        }
+        catch (NessieNamespaceNotEmptyException e) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR, format(
+                    "Namespace '%s' is not empty. One or more tables exist", namespace), e);
+        }
+    }
+
+    @Override
+    public Map<String, Object> loadNamespaceMetadata(ConnectorSession session, String namespace)
+    {
+        try {
+            nessieClient.getNessieApi().getNamespace()
+                    .refName(nessieClient.getRef().getName())
+                    .namespace(namespace)
+                    .get();
+        }
+        catch (NessieNamespaceNotFoundException e) {
+            throw new SchemaNotFoundException(namespace);
+        }
+        catch (NessieReferenceNotFoundException e) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR,
+                    format("Cannot load Namespace '%s': ref is no longer valid", namespace),
+                    e);
+        }
+        return propertiesByNamespace.get(namespace);
+    }
+
+    @Override
+    public Optional<TrinoPrincipal> getNamespacePrincipal(ConnectorSession session,
+            String namespace)
+    {
+        return Optional.empty();
+    }
+
+    @Override
+    public void createNamespace(ConnectorSession session, String namespace,
+            Map<String, Object> properties, TrinoPrincipal owner)
+    {
+        try {
+            nessieClient.getRef().checkMutable();
```

Comment:
These two calls : 

- check mutable
- create namespace

are implementation details related to the nessie client.
The catalog should just provide the required parameters:

- the reference name
- the namespace 

and the client should take care of the rest.

Reply:
sgtm